### PR TITLE
Unit testing

### LINKED ISSUE #28: [Packaging | Unitary tests]
- [ ] Get familiarised with unitary tests.
    - You can get inspired of any formal repository (eOSSR has some examples too)
 - [ ] Implement some basic tests for the example scripts that you have already created.

### PR DESCRIPTION
**Pull request by mrzengel**
Developed unit testing of basic python functions + incorporated into build workflow (TBD about eventually adding the badge to the README)

Closes #28

### CODE REVIEW

**Review by garciagenrique — COMMENTED** [review]
lgtm, I would like to know @Soap2G thoughts :)

File: zenodo_jupyterlab/server/tests/test_testConnection.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+@pytest.mark.asyncio
+@patch('zenodo_jupyterlab.server.testConnection.ZenodoAPI')
+async def test_zenodo_connection_success(MockZenodoAPI):
+    # Mock the ZenodoAPI instance and its method
+    mock_instance = MockZenodoAPI.return_value
+    mock_instance.query_user_deposits = Mock()
+    mock_instance.query_user_deposits.return_value.status_code = 200
+
+        # Mock the environment variable
+    with patch.dict(os.environ, {'ZENODO_API_KEY': 'fake_correct_api_key'}):
```
Comment (Soap2G) [inline zenodo_jupyterlab/server/tests/test_testConnection.py:16]: Could we set up an ENV variable on github to point to the Zenodo sandbox? Let's call it `ZENODO_API_KEY`.
I'd rather use ENV vars than `'fake_correct_api_key'`.

File: zenodo_jupyterlab/server/tests/test_testConnection.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+@pytest.mark.asyncio
+@patch('zenodo_jupyterlab.server.testConnection.ZenodoAPI')
+async def test_zenodo_connection_failure(MockZenodoAPI):
+    # Mock the ZenodoAPI instance to raise an exception
+    mock_instance = MockZenodoAPI.return_value
+    mock_instance.query_user_deposits = Mock()
+    mock_instance.query_user_deposits.side_effect = Exception('Failed')
+
+    # Mock the environment variable
+    with patch.dict('os.environ', {'ZENODO_API_KEY': 'fake_false_api_key'}):
```
Comment (Soap2G) [inline zenodo_jupyterlab/server/tests/test_testConnection.py:34]: Same thing here.

**Review by Soap2G — CHANGES_REQUESTED** [review]

File: zenodo_jupyterlab/server/tests/test_testConnection.py (hunk 2 above)
Comment (mrzengel) [reply zenodo_jupyterlab/server/tests/test_testConnection.py:34]: or at least I can't access the link you sent nor can I find it in the settings

**Review by Soap2G — APPROVED** [review]
lgtm; godspeed.

**Pull request merged**